### DISCUSSION TITLE
VerneMQ re-validate client's certificate for expiry as sessions are persistent

### DISCUSSION THREAD

**Discussion by vshirol** [vernemq/vernemq#1990] [Q&A]
We are planning to use TLS based communication between VerneMQ server and MQTT clients connecting to it. Below are our questions:

 •	Is there a way to periodically re-validate the client certificates, for example on a daily basis? (I didn't find anything related to this in the MQTT spec. I only know that there is the keep alive with PINGREQ and PINGRESP, but nothing related to a periodical re-validation of certificates.)
•	How could such a periodical certificate re-validation every 24 hours be done? 
•	What happens if there is a persistent MQTT session established for several months but the client certificate has expired in the meantime and is not valid anymore, will the session just be kept alive as long as possible?

**Comment by tobdub-snce** [comment]
As I understand it there is no official way to re-validate.
I have implemented something similar in my auth plugin by disconnecting the clients using the admin api when authentication expires. The clients then automatically reconnect and authentication is validated.

**Comment by ioolkos** [comment]
MQTT v5 has a reauthentication feature which is based on a reauth request by the client not the server:
https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Re-authentication

Verne can reauthorize (not reauthenticate) sessions (with the `vmq-admin session reauthorize` command). This will currently only reauthorize subscriptions (by pushing them through the plugin hooks again) but not Publish ACLs.

**Reply by vshirol** [reply]
Thanks @tobdub and @ioolkos, I will explore more on both of these options.
